com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -14,19 +14,105 @@
 using MLAPI.Profiling;
 using MLAPI.Reflection;
 using MLAPI.Security;
+using MLAPI.Serialization;
 using MLAPI.Serialization.Pooled;
 using MLAPI.Spawning;
 using BitStream = MLAPI.Serialization.BitStream;
-using MLAPI.Serialization;
 using Unity.Profiling;
 
+#if UNITY_EDITOR
+using System.Runtime.CompilerServices;
+[assembly: InternalsVisibleTo("Unity.Multiplayer.MLAPI.Editor.CodeGen")]
+#endif // UNITY_EDITOR
+
 namespace MLAPI
 {
     /// <summary>
     /// The base class to override to write networked code. Inherits MonoBehaviour
     /// </summary>
     public abstract partial class NetworkedBehaviour : MonoBehaviour
     {
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal enum NExec
+        {
+            None = 0,
+            Server = 1,
+            Client = 2
+        }
+
+        const String StandardRPC_Channel = "STDRPC";
+
+#pragma warning disable 414
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal NExec __nexec = NExec.None;
+#pragma warning restore 414
+
+        // RuntimeAccessModifiersILPP will make this `protected`
+        internal BitWriter BeginSendServerRpc(ServerRpcSendParams sendParams, bool isReliable)
+        {
+            // @mfatihmar (Unity) Begin: Temporary, placeholder implementation
+            RPCQueueManager rpcQueueMananger = NetworkingManager.Singleton.GetRPCQueueManager();
+            if(rpcQueueMananger != null)
+            {
+                var writer = rpcQueueMananger.BeginAddQueueItemToOutboundFrame(RPCQueueManager.QueueItemType.ServerRPC, Time.realtimeSinceStartup, StandardRPC_Channel,0, NetworkingManager.Singleton.ServerClientId,null);
+
+                writer.WriteBit(false); // Encrypted
+                writer.WriteBit(false); // Authenticated
+                writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
+                writer.WriteUInt64Packed(NetworkId); // NetworkObjectId
+                writer.WriteUInt16Packed(GetBehaviourId()); // NetworkBehaviourId
+
+                return writer;
+            }
+            return null;
```

Comment:
Is this an error condition?  Like would crashing here actually be the right thing?

Reply:
Yes, the returning of the null pointer tells the ILPP Codegen side that there is no writer to use as there is no RPCQueueManager (what creates the RPCQueueHistoryFrames that hold a writer to each respective stream) to get a writer from.  If the ILPP Codgen'd side detects no writer then it exits too (as it has nothing to write with).
So, it is a way to let that portion of the data flow pipeline know that something is "not right" and to exit the RPC generation process.